### PR TITLE
CSHARP-3178: MongoDB.Driver latest version 2.11 breaks compiling Xamarin iOS.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**
EG: https://evergreen.mongodb.com/version/5f75c1653066155bcb1dd6f7

### CODE REVIEW

File: src/MongoDB.Driver.Core/MongoDB.Driver.Core.csproj [hunk 1]
```diff
@@ -61,7 +61,7 @@
     <SnappyBinaries>Core/Compression/Snappy/lib/win</SnappyBinaries>
   </PropertyGroup>
 
-  <ItemGroup>
+  <ItemGroup Condition=" '$(OS)' == 'Windows_NT' ">
```
Comment (jyemin) [inline src/MongoDB.Driver.Core/MongoDB.Driver.Core.csproj]: Didn't the report say that it worked with 2.10 release, which already had Snappy?

Comment (DmitryLukyanov) [reply src/MongoDB.Driver.Core/MongoDB.Driver.Core.csproj]: not really. The snappy with native libraries was added only in 2.11. See here: https://jira.mongodb.org/browse/CSHARP-2966

Comment (DmitryLukyanov) [reply src/MongoDB.Driver.Core/MongoDB.Driver.Core.csproj]: We checked and this issue also happens with snappy. The reason why the reporter doesn't see it, mostlikely is related to the fact that `vs for mac` shows only the first thrown error

**Comment by DmitryLukyanov** [comment]
I temporarily removed reviewers to check failed EG tasks

File: src/MongoDB.Driver.Core/MongoDB.Driver.Core.csproj [hunk 2]
```diff
@@ -25,6 +25,18 @@
     <IncludeSymbols>true</IncludeSymbols>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <IsWindows Condition="$([MSBuild]::IsOSPlatform('Windows'))">true</IsWindows>
+    <IsOSX Condition="$([MSBuild]::IsOSPlatform('OSX'))">true</IsOSX>
+    <IsLinux Condition="$([MSBuild]::IsOSPlatform('Linux'))">true</IsLinux>
+  </PropertyGroup>
+
+  <Target Name="PrintOS" BeforeTargets="Build">
```
Comment (DmitryLukyanov) [inline src/MongoDB.Driver.Core/MongoDB.Driver.Core.csproj:34]: lines 34-38 are used only for logging. In windows, you will see the used values in the build output. Can be removed without any problems :)

File: src/MongoDB.Driver.Core/MongoDB.Driver.Core.csproj [hunk 3]
```diff
@@ -25,6 +25,18 @@
     <IncludeSymbols>true</IncludeSymbols>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <IsWindows Condition="$([MSBuild]::IsOSPlatform('Windows'))">true</IsWindows>
```
Comment (DmitryLukyanov) [inline src/MongoDB.Driver.Core/MongoDB.Driver.Core.csproj:29]: The main change here where we determine whether the current platform is windows or no

Comment (DmitryLukyanov) [reply src/MongoDB.Driver.Core/MongoDB.Driver.Core.csproj:29]: In the first version of this pr, I used ` <ItemGroup Condition=" '$(OS)' == 'Windows_NT' ">`: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/152/commits/7ebafd77fa2f6d25f134a7c01543376a987f5dba#diff-5753749920fdae3d4731dacc1375a126R64 
which should work, but for some reason, the `OS` value on EG is `windows-64`. We definitely can add a condition like: `OS == Windows_NT or OS = Windows-64`, but I'm not sure how many similar aliases can be used here. So, I think it's possible that we can miss something in this approach

**Comment by DmitryLukyanov** [comment]
Waiting for @durran updates :)

**Comment by DmitryLukyanov** [comment]
Now, the user won't have a problem with app compiling on non-windows `OS`s. But he will still have an error if he tries to configure compression directly. 
It will be either the FileNotFound exception(compression libraries are presented only on windows) or PlatformNotSupportedException in case if he uses `targetFramework` different than `netstandard1.5` (we addressed this issue here CSHARP-3238)

cc: @durran , @jyemin

**Review by durran — APPROVED** [review]
LGTM - This worked on my machine.

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request closed without merging**